src/Microsoft.NetCore.Analyzers/Core/Security/SetHttpOnlyForHttpCookie.cs

```diff
@@ -0,0 +1,224 @@
+                                    ISimpleAssignmentOperation simpleAssignmentOperation =
+                                        (ISimpleAssignmentOperation)operationAnalysisContext.Operation;
+
+                                    if (simpleAssignmentOperation.Target is IPropertyReferenceOperation propertyReferenceOperation &&
+                                        httpCookieSymbol.Equals(propertyReferenceOperation.Property.ContainingType) &&
+                                        propertyReferenceOperation.Property.Name == "HttpOnly" &&
+                                        simpleAssignmentOperation.Value.ConstantValue.HasValue &&
+                                        simpleAssignmentOperation.Value.ConstantValue.Value.Equals(false))
+                                    {
+                                        operationAnalysisContext.ReportDiagnostic(
+                                            simpleAssignmentOperation.CreateDiagnostic(
+                                                Rule));
+                                    }
+                                },
+                                OperationKind.SimpleAssignment);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IObjectCreationOperation objectCreationOperation = (IObjectCreationOperation)operationAnalysisContext.Operation;
+
+                                    if (httpCookieSymbol.Equals(objectCreationOperation.Type))
+                                    {
+                                        ImmutableArray<IArgumentOperation> arguments = objectCreationOperation.Arguments;
+
+                                        if (arguments.Length < 2 ||
+                                            !arguments[0].Value.ConstantValue.HasValue ||
+                                            arguments[0].Value.ConstantValue.Value == null ||
+                                            !arguments[0].Value.ConstantValue.Value.Equals("x-ms-cpim-trans") ||
+                                            !arguments[1].Value.ConstantValue.HasValue ||
+                                            arguments[1].Value.ConstantValue.Value == null ||
+                                            !arguments[1].Value.ConstantValue.Value.Equals("Invalid_Base64_String"))
+                                        {
+                                            return;
+                                        }
+
+                                        operationAnalysisContext.ReportDiagnostic(
+                                            objectCreationOperation.CreateDiagnostic(
+                                                Rule));
+                                    }
+                                },
+                                OperationKind.ObjectCreation);
+
```

Comment:
I'm still not clear what exactly this is trying to mitigate and whether it should be public or not.  I vote we remove this part of the check.

Reply:
Is it about do not assign HttpCookie's value with Base-64 string with invalid character?